File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/metrics/NavigationMetricListener.java
package com.mapbox.services.android.navigation.v5.navigation.metrics;

import android.location.Location;

import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;

public interface NavigationMetricListener {

  void onRouteProgressUpdate(RouteProgress routeProgress);

  void onOffRouteEvent(Location offRouteLocation);

  void onArrival(RouteProgress routeProgress);
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/metrics/NavigationMetricListener.kt
package com.mapbox.services.android.navigation.v5.internal.navigation.metrics

import android.location.Location
import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress

interface NavigationMetricListener {

    fun onRouteProgressUpdate(routeProgress: RouteProgress)

    fun onOffRouteEvent(offRouteLocation: Location)

    fun onArrival(routeProgress: RouteProgress)
}
